src/App.tsx

```diff
@@ -1,26 +1,253 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useEffect, useRef, useState } from 'react';
+
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+import { Footer } from './Components/Footer';
+import { ErrorNotification } from './Components/ErrorNotification';
+import { Header } from './Components/Header';
+import { TodoList } from './Components/TodoList';
+import { TodoItem } from './Components/TodoItem';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isAdding, setIsAdding] = useState(false);
+  const [deletingIds, setDeletingIds] = useState<number[]>([]);
+  const [updatingIds, setUpdatingIds] = useState<number[]>([]);
```

Comment:
You can keep only one state for it and use for both operations. But rename to `loadingTodoIds`